src/main/java/impl/weather/YearTemperatureStatsimpl.java

```diff
@@ -0,0 +1,83 @@
+package impl.weather;
+
+import api.weather.DayTemperatureInfo;
+import api.weather.YearTemperatureStats;
+import kotlin.Pair;
+
+import java.awt.*;
+import java.time.Month;
+import java.util.*;
+import java.util.List;
+
+public class YearTemperatureStatsimpl implements YearTemperatureStats {
+    private HashMap<Month, HashMap<Integer, DayTemperatureInfo>> status = new HashMap<>();
+    private HashMap<Month, Pair<Double, Integer>> averageTemperature = new HashMap<>();
```

Comment:
Please, try not to use Pair. It is better to create separate class which present such data